### PR TITLE
Different changes, see PR #58

### PR DESCRIPTION
**Pull request by Yardanico**
Almost all of this is described in #58 , I tried to make separate commits.
I think that after reviewing it's better to do a squash merge so that this "revert" commit doesn't get into history

### CODE REVIEW

**Comment by Yardanico** [comment]
This is almost the same stuff as I had in #58 but without 4->2 indent (it can be done in a separate PR). I may have introduced some new bugs but I tried to not alter the program logic (but I simplified it a bit in some places)

**Comment by 0atman** [comment]
This is an amazing PR, thank you so much @Yardanico!

I'll get a few more PRs merged today and then cut a release :muscle: 

Thanks @Tangdongle for the review too

**Comment by timotheecour** [comment]
@Yardanico I believe this is causing https://github.com/inim-repl/INim/issues/63

**Comment by 0atman** [comment]
I merged without running `nimble test`, I'm a fraud

**Comment by timotheecour** [comment]
haha no worries; it's all about setting up automation to remove human errors; no one should have to manually run `nimble test` before merging :-)

File: src/inim.nim [hunk 1]
```diff
@@ -229,108 +235,102 @@ proc getPromptSymbol(): string =
...
-    app.new()
-    app.nim = "nim"
-    app.srcFile = ""
-    app.showHeader = true
+    app = App(
+        nim: nim,
+        srcFile: srcFile,
+        showHeader: showHeader,
+        flags: flags
+    )
```
Comment (0atman) [inline src/inim.nim:326]: @Yardanico This refactor broke the tests. I've opened up https://github.com/inim-repl/INim/issues/64, would you have some time to fix the test?

**Pull request merged**